org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/ExecuteView.java

```diff
@@ -643,19 +649,19 @@ private String getDetectorNames() throws Exception {
     	return detectors;
 	}
 
-	private List<String> getMonitors() throws Exception {
+	private List<String> getMonitors(MonitorRole monitorRole) throws Exception {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)